### PR TITLE
Parse final closure date also

### PR DESCRIPTION
**Pull request by paulius-pilkauskas**
Parse"final closure date" field also.

@vinted/shipping-backend

### CODE REVIEW

**Comment by zverseckas** [comment]
What does this field mean?

**Comment by paulius-pilkauskas** [comment]
> What does this field mean?

The `final closure date` is the final closure date, it’s a specific field which means that the parcel shop is definitely closed.

I had a conversation with Phillipe from MR and he explained this field.

File: lib/mondial_relay/parcel_shops/fetch_all/parse_line.rb [hunk 1]
```diff
@@ -104,6 +106,23 @@ def opening_hours(opens_at_position, closes_at_position)
...
+
+          starts_at = Date.parse(ends_at) - 1
+
+          {
+            starts_at: starts_at.strftime(DATE_FORMAT),
+            ends_at: ends_at,
+          }
+        end
+
         def holiday_information
```
Comment (mangirdasvinted) [inline lib/mondial_relay/parcel_shops/fetch_all/parse_line.rb:126]: This method seems to be unused ?

File: lib/mondial_relay/parcel_shops/fetch_all/parse_line.rb [hunk 2]
```diff
@@ -104,6 +106,23 @@ def opening_hours(opens_at_position, closes_at_position)
...
+        def closure_periods
+          holiday_information.push(final_closure_date).compact
+        end
+
+        def final_closure_date
+          ends_at = line.slice(72, 10).strip
+
+          return if ends_at.blank?
+
+          starts_at = Date.parse(ends_at) - 1
```
Comment (martynasdub) [inline lib/mondial_relay/parcel_shops/fetch_all/parse_line.rb:118]: Move `1` to a meaningul named constant, and make it `1.day`

Comment (paulius-pilkauskas) [reply lib/mondial_relay/parcel_shops/fetch_all/parse_line.rb:118]: 1.day is not working here:
```
  1) MondialRelay::ParcelShops::FetchAll::ParseLine.for with a final closure date parses final closure date
     Failure/Error: starts_at = Date.parse(ends_at) - 1.day

     NoMethodError:
       undefined method `day' for 1:Integer
```

**Comment by zverseckas** [comment]
@paulius-pilkauskas, thanks for explaining. So if this parcel shops closes at `starts_at` and you set its `ends_at` is `starts_at + 1 day` how does it work? It looks like this point is closed for one day? How are you going to use it?

Have you thought about adding some new field like `permanently_closes_on` or something. So that you can disable this point in advance.

**Comment by paulius-pilkauskas** [comment]
> @paulius-pilkauskas, thanks for explaining. So if this parcel shop closes at `starts_at` and you set its `ends_at` to `starts_at + 1 day` how does it work? It looks like this parcel shop is closed for one day? How are you going to use it?
> 
> Have you thought about adding some new field like `permanently_closes_on` or something. So that you can disable this point in advance.

The algorithm is a little bit different. It sets the `starts_at` to `ends_at - 1 day`.
The pros of this solution is that we don't need additional changes in adapters. The lib will return an extra additional closure period. Moreover, I think that this solution looks OK, it is a valid closure period with the one day day off :)

I don't like new field like `permanently_closes_on` due to it requires additional changes from the `svc-shipping` side. However, this new field has some advantages like you mentioned - the parcel shop could be deactivated and it will be skipped during fetch nearby action.

I am thinking about to dig the secondary solution. I will back to you with some insights and possible solutions.

**Pull request merged**